### PR TITLE
Feature/fix ard calc

### PR DESCRIPTION
**Pull request by drezap**
#### Submission Checklist

- [x] Run unit tests: `./runTests.py test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary:
Add a seperate length scale for each dimension in the prim's exponential quadratic kernel. The calculation first takes in the vector of vectors, and divides each dimension by each length scale. This was suggested by Michael Riis Anderson.

Also uploaded docs to include formulas, better descriptions, and included the word "magnitude" for the "marginal standard deviation" which has more intersection with the GP literature. 

I've done simulations to verify accuracy, with stan's docs and other GP software. Apologies, I'll do more extensive testing from this point on (more than just different datatypes).

#### How to Verify:
Simulations, test cases.

#### Side Effects:
N/A

#### Documentation:
Updated doxygen to include formulas and 
#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by drezap** [comment]
this is same issue as #921 I'm going to reset and hope it hops to another machine. If I can get access to this machine I'm happy to fix the MPI config...

**Comment by drezap** [comment]
can some one review this please? @syclik sorry to ping you, but I think you're the guy!

**Comment by syclik** [comment]
Can you summarize the changes to the code? What are the new function signatures and which ones changed?

File: stan/math/prim/mat/fun/cov_exp_quad.hpp [hunk 1]
```diff
@@ -66,75 +74,86 @@ inline
...
+ * @throw std::invald_arguemnt if dimension of x is not the same
+ *   length as length_scale
  */
 template <typename T_x, typename T_sigma, typename T_l>
-inline
-    typename Eigen::Matrix<typename stan::return_type<T_x, T_sigma, T_l>::type,
-                           Eigen::Dynamic, Eigen::Dynamic>
-    cov_exp_quad(const std::vector<T_x>& x, const T_sigma& sigma,
-                 const std::vector<T_l>& length_scale) {
+inline typename boost::enable_if_c<
```
Comment (syclik) [inline stan/math/prim/mat/fun/cov_exp_quad.hpp]: Why is this `enable_if`'d?

Comment (syclik) [reply stan/math/prim/mat/fun/cov_exp_quad.hpp]: If this is limiting `T_x`, then please update the doc appropriately for `@tparam T_x`.

File: stan/math/prim/mat/fun/cov_exp_quad.hpp [hunk 2]
```diff
@@ -182,45 +207,67 @@ cov_exp_quad(const std::vector<T_x1>& x1, const std::vector<T_x2>& x2,
...
+ * @throw std::invalid_argument if dimension of x1 or dimension of x2 is not the
+ *   same length as the length scale
  */
 template <typename T_x1, typename T_x2, typename T_sigma, typename T_l>
-inline typename Eigen::Matrix<
-    typename stan::return_type<T_x1, T_x2, T_sigma, T_l>::type, Eigen::Dynamic,
-    Eigen::Dynamic>
-cov_exp_quad(const std::vector<T_x1>& x1, const std::vector<T_x2>& x2,
-             const T_sigma& sigma, const std::vector<T_l>& length_scale) {
+inline typename boost::enable_if_c<
```
Comment (syclik) [inline stan/math/prim/mat/fun/cov_exp_quad.hpp]: Why is this `enable_if` only check `T_x1` and not also `T_x2`?

File: stan/math/prim/mat/fun/cov_exp_quad.hpp (hunk 1 above)
Comment (drezap) [reply stan/math/prim/mat/fun/cov_exp_quad.hpp]: I enabled if'd it because having seperate length scales for each dimension is only possible when D > 1, which is only possible if T_x in this case is a vector like. Is this necessary? I think then the function will only be called if length scale is a vector, so the function call is unique, so I think it's fine. I'm just removing all the enable ifs

**Comment by drezap** [comment]
@syclik no new function signatures, I modified the ones with `std::vector<T_l> length_scale`, that is, where l is a vector. I just divided each dimension by each length scale, prior calculating the kernel.

**Comment by drezap** [comment]
Going to apply everything daniel mentioned on #867 to all of these

**Comment by drezap** [comment]
closing until some edits are made, not ready for review
